### PR TITLE
added filter

### PR DESCRIPTION
**Pull request by grepoz**
Added filter that uses Morfeusz2 module.
Filter:
- preprocess text
- check if word is in badwords dict
- check if Morfeusz2 classified word as vulgarism

Check out TextPreprocessor class to not duplicate code

### CODE REVIEW

**Comment by jereml99** [comment]
Please rebase on main

**Comment by SunBear1** [comment]
CONFLICTS. RESOLVE THEM. NOW.

File: src/filters.py [hunk 1]
```diff
@@ -62,7 +69,22 @@ def remove_repeats(word: str) -> str:
...
+        word_core = analysis[0][2][1]
+        if word_core == "":
+            raise Exception
+    except:
+        print("enable to remove ending")
+
+    return extract_lemma(word_core)
+
+
+def extract_lemma(_word):
```
Comment (SunBear1) [inline src/filters.py:85]: This function is unnecessary in my opinion. You can implement these 2 lines inside `remove_endings` function.

File: src/filters.py [hunk 2]
```diff
@@ -1,12 +1,18 @@
 from itertools import product
 
 from source_dicts import SourceDictionaries
+from itertools import product, combinations
```
Comment (FreddieHorn) [inline src/filters.py:4]: doubled itertools import

File: src/filters.py [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
+import morfeusz2
+
 
 """
 Module containing all micro filers for word validation
 """
 
 
+morf = morfeusz2.Morfeusz()
```
Comment (SunBear1) [inline src/filters.py:13]: Why is morf global if only one function uses this variable?

File: src/filters.py [hunk 4]
```diff
@@ -62,7 +69,22 @@ def remove_repeats(word: str) -> str:
...
-    return word
+    analysis = morf.analyse(word)
+
+    word_core = ""
+    try:
+        word_core = analysis[0][2][1]
+        if word_core == "":
+            raise Exception
+    except:
+        print("enable to remove ending")
```
Comment (SunBear1) [inline src/filters.py:80]: Our CLI is in polish, please translate the error msg

**Review by SunBear1 — COMMENTED** [review]
Please add some tests to the E2E tests, because they inform us how good is our filter. 

https://github.com/SunBear1/SigmaFilter/blob/fe3c3b98766127bb83504c29f46306b006cbce16/test/e2e/test_filtering.py#L6

File: test/unit/test_filters.py [hunk 5]
```diff
@@ -1,9 +1,18 @@
-from src.filters import remove_repeats, remove_special_characters
+from src.console_utils import filter_badwords, filter_badwords_adjacent_words
+from src.filters import remove_repeats, remove_endings
+import unittest
```
Comment (FreddieHorn) [inline test/unit/test_filters.py:4]: I dont think we are sticking to 'src.filename' convention as we can mark src folder as a root.  @SunBear1  should confirm this

**Review by FreddieHorn — CHANGES_REQUESTED** [review]

File: test/unit/test_filters.py (hunk 5 above)
Comment (SunBear1) [reply test/unit/test_filters.py:4]: By marking `src` directory as `Sources root` we don't have to use src prefix. Please refactor our code under this condition.

File: test/unit/test_filters.py [hunk 6]
```diff
@@ -1,9 +1,19 @@
...
-def test_remove_special_characters():
-    assert remove_special_characters("ci.pa") == "cipa"
+    def test_removingEnding_whenWordIsNoun_thenReturnsWordCore(self):
+        self.assertEqual("kurwa", remove_endings(self.morf, "kurwę"))
+
+    def test_removingEnding_whenWordIsAdjective_thenReturnsWordCore(self):
+        self.assertEqual("jebać", remove_endings(self.morf, "jebana"))
+
+    def test_removingEnding_whenWordIsAdverb_thenReturnsWordCore(self):
+        self.assertEqual("chujowy", remove_endings(self.morf, "chujowa"))
```
Comment (jereml99) [inline test/unit/test_filters.py:19]: Use pytest please

**Review by jereml99 — APPROVED** [review]

File: src/filters.py [hunk 7]
```diff
@@ -61,10 +63,35 @@ def remove_repeats(word: str) -> str:
...
+    word = remove_diminutive(word)
+    word = remove_adjective_ends(word)
+    word = remove_verbs_ends(word)
+    word = remove_adverbs_ends(word)
+    word = remove_plural_forms(word)
+    word = remove_general_ends(word)
     return word
 
 
+def remove_endings(analyzer, word) -> str:
```
Comment (SunBear1) [inline src/filters.py:77]:
```suggestion
def remove_endings(analyzer: <whatever is morfeusz type>, word: str) -> str:
```

File: src/remove_endings_helper.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
+# taken from https://github.com/Tutanchamon/pl_stemmer/blob/master/pl_stemmer.py
+def remove_general_ends(word):
```
Comment (SunBear1) [inline src/remove_endings_helper.py:2]:
```suggestion
def remove_general_ends(word: str) -> str:
```

Comment (SunBear1) [reply src/remove_endings_helper.py:2]: Apply this to other functions

File: test/e2e/test_filtering.py [hunk 9]
```diff
@@ -70,13 +71,27 @@
...
     ("<hjo", 4),
-    ("chhoooy" ,4),
+    ("chhoooy", 4),
     ("pindoi", 6),
     ("pllndl0", 6),
     ("pind?ol", 6),
     ("piiiiiiiiind0i", 6),
     ("€¥€k0m@n", 8),
     ("curva", 5),
+
```
Comment (SunBear1) [inline test/e2e/test_filtering.py:81]:
```suggestion
```

**Review by SunBear1 — APPROVED** [review]
lgtm 👑

**Review by FreddieHorn — APPROVED** [review]

**Pull request merged**